include/boost/geometry/algorithms/detail/closest_feature/point_to_range.hpp

```diff
@@ -72,8 +72,10 @@ class point_to_point_range
         for (++prev, ++it; it != last; ++prev, ++it)
         {
             Distance dist = strategy.apply(point, *prev, *it);
-            if (geometry::math::equals(dist, zero))
-            {
+
+            // Stop only if we find exactly zero distance
+            // otherwise it may stop at some very small value and miss the min
+            if (dist == zero){
```

Comment:
The curly brace should go to next line

Reply:
The `dist` can probably be const (this was already but maybe you can change it on the go)
`Distance const dist = ...`